packages/cli/__tests__/daemon/__integration__/enable/cli.daemon.enable.integration.test.ts

```diff
@@ -0,0 +1,301 @@
+/*
+* This program and the accompanying materials are made available under the terms of the
+* Eclipse Public License v2.0 which accompanies this distribution, and is available at
+* https://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Copyright Contributors to the Zowe Project.
+*
+*/
+
+import * as fs from "fs";
+import * as nodeJsPath from "path";
+import * as which from "which";
+
+import { IO, ISystemInfo, ProcessUtils } from "@zowe/imperative";
+import { ITestEnvironment, runCliScript } from "@zowe/cli-test-utils";
+
+import { TestEnvironment } from "../../../../../../__tests__/__src__/environment/TestEnvironment";
+import { ITestPropertiesSchema } from "../../../../../../__tests__/__src__/properties/ITestPropertiesSchema";
+
+let testEnvironment: ITestEnvironment<ITestPropertiesSchema>;
+
+describe("daemon enable", () => {
+    const rimraf = require("rimraf").sync;
+    const fakeExeContent = "This is not a real executable";
+    const exeCantRunDaemonMsg1: string = "You cannot run this 'daemon' command while using the Zowe-CLI native executable.";
+    const exeCantRunDaemonMsg2: string = "Copy and paste the following command instead:";
+    const EXIT_CODE_CANT_RUN_DAEMON_CMD: number = 107;
+
+    let exePath: string;
+    let pathToBin: string;
+    let preBldTgzPath: string;
+    let isZoweExe: boolean = false; // is the zowe command that we will run an executable?
+
+    beforeAll(async () => {
+        // Create the unique test environment
+        testEnvironment = await TestEnvironment.setUp({
+            testName: "daemon_enable_integration",
+            skipProperties: true
+        });
+
+        // determine our current OS
+        const sysInfo: ISystemInfo = ProcessUtils.getBasicSystemInfo();
+
+        // form our tgz file name
+        let tgzFileName = "zowe-";
```

Comment:
Timothy is correct. These are fake TGZs used to test that "daemon enable" forms the right file name for every OS.

However, that does bring up another good point, The integration tests look for real TGZs in a "prebuilds" directory that is a sibling of the cli/lib directory. Those TGZs will exist after an NPM install of zowe-cli. However, they won't exist during pipeline tests run from source code. We will either have to disable those integration tests, or create the TGZs during the pipeline. Creating TGZs may be problematic, since the zowe.exe is not always built from source in every pipeline. I think that is done in GitHub Actions, but not Jenkins.

Reply:
Ignore my last comment about real TGZs being needed for integration tests. The tests are all written so that even when the `daemon enable` command tries to run a fake executable to get its version number, the test matches enough text to confirm that the operation was properly attempted, even though the fake EXE bombed.

The only place where a test actually requires a real EXE is in cli.zowe.exe.integration.test.ts. It will use the EXE from the zowex/target/release-or-debug/zowe[.exe] file, which the GitHub Action for `Rust CLI CI` builds. If that file does not exist, the test script skips that test so that the test script succeeds on pipelines that do not build the EXE.

We will see if this actually works in the pipelines, after we resolve other unrelated pipeline problems.